File: frontend/angular-ecommerce/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
    imageUrl:string;
    price: number;
    quantity: number;
    bookId: number;


    constructor(cartItem: CartItem){
        this.imageUrl = cartItem.imageUrl;
        this.price = cartItem.price
        this.quantity = cartItem.quantity;
        this.bookId = cartItem.id;
        console.log(this.bookId)
    }
}

File: backend/ecommerce/src/main/java/com/ez_lib/ecommerce/entity/OrderItem.java
package com.ez_lib.ecommerce.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;



@Entity
@Table(name="order_item")
@Getter
@Setter
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;

    @Column(name="image_url")
    private String imageUrl;

    @Column(name="price")
    private BigDecimal unitPrice;

    @Column(name="book_id")
    private Long bookId;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;

}